src/stan/analyze/mcmc/split_chains.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef STAN_ANALYZE_MCMC_SPLIT_CHAINS_HPP
+#define STAN_ANALYZE_MCMC_SPLIT_CHAINS_HPP
+
+#include <cmath>
+#include <vector>
+#include <algorithm>
+
+namespace stan {
+namespace analyze {
+
+  /**
+   * Returns samples split in half.
```

Comment:
Perhaps "Splits each chain into two chains of equal length."?  Also as above what happens when the chain length is odd?